### PR TITLE
Add Vendor Authorized Boot support to HPS boot image

### PR DESCRIPTION
**Pull request by anandaravuri**
- add  --vab and  --quartus  commands to make bake
       - copy userkey_root_public.qky, userkey_root_private.pem, userkey_root_public.pem certs to VAB folder
       - VAB Make bake build process steps
            - Quartus _sign tool signs UBoot and Linux images (bl31.bin, u-boot-nodtb.bin, u-boot.dtb, linux.dtb, rootfs.cpio, Image)
            - Creates Vendor Authorized Boot HPS image  u-boot-userkey-vab.itb and  Hex file u-boot-spl-dtb-vab.hex

### CODE REVIEW

File: tools/meta-bake/meta-bake.py [hunk 1]
```diff
@@ -249,6 +247,161 @@ def make_fit(self, target, ctx: dict):
...
+        print(f'uboot_make_dir={uboot_make_dir}')
+
+        # copy uboot src to folder uboot-socfpga-vab-n6000
+        shutil.copytree(uboot_src_dir, uboot_vab)
+
+        # quartus path
+        quartus_path = os.path.abspath('{}/{}'.format(args.quartus,
+                                                      "bin/quartus_sign"))
+        if not os.path.exists(quartus_path):
+            print(" Invalid quartu_path:", quartus_path)
```
Comment (tswhison) [inline tools/meta-bake/meta-bake.py]: s/quartu_path/quartus_path/

File: tools/meta-bake/meta-bake.py [hunk 2]
```diff
@@ -249,6 +247,161 @@ def make_fit(self, target, ctx: dict):
...
+
+        # Sign uboot binary
+        for i in range(len(uboot_binary)):
+            print(uboot_binary[i])
+
+            # copy unsinged vab files
+            shutil.copy(uboot_make_dir.joinpath(uboot_binary[i]), build_vab)
+
+            # delete unsigned_cert.cert
+            unsigned_cert_path = build_vab + "/" + "unsigned_cert.ccert"
```
Comment (tswhison) [inline tools/meta-bake/meta-bake.py:338]: Is unsigned_cert.ccert correct or should it be unsigned_cert.cert?

File: tools/meta-bake/meta-bake.py [hunk 3]
```diff
@@ -249,6 +247,161 @@ def make_fit(self, target, ctx: dict):
...
+            print("unsigned_cert_patht:", unsigned_cert_path)
```
Comment (tswhison) [inline tools/meta-bake/meta-bake.py]: s/unsigned_cert_patht/unsigned_cert_path/

File: tools/meta-bake/meta-bake.py [hunk 4]
```diff
@@ -249,6 +247,161 @@ def make_fit(self, target, ctx: dict):
...
+            if os.path.exists(unsigned_cert_path):
+                os.remove(unsigned_cert_path)
+
+            # create unsigned_cert.ccert
+            p1 = subprocess.Popen(args=[fcs_prepare,
```
Comment (tswhison) [inline tools/meta-bake/meta-bake.py]: Bandit recommends against using subprocess.Popen.  Use subprocess.check instead.

File: tools/meta-bake/meta-bake.py [hunk 5]
```diff
@@ -249,6 +247,161 @@ def make_fit(self, target, ctx: dict):
...
+                                  '--hps_cert', uboot_binary[i], '-v'], cwd=build_vab)
+            p1.wait()
+
+            # create signed cert
+            signed_cert = 'signed_cert{}.ccert'.format(uboot_binary[i])
```
Comment (tswhison) [inline tools/meta-bake/meta-bake.py:348]: Is .ccert correct or should it be .cert?

File: tools/meta-bake/meta-bake.py (hunk 2 above)
Comment (anandaravuri) [reply tools/meta-bake/meta-bake.py:338]: ../fcs_prepare --hps_cert ../uboot-socfpga/u-boot-dtb.img -v 
The resulting certificate is in unsigned_cert.ccert and we need to sign it
ccert is correct

File: tools/meta-bake/meta-bake.py (hunk 5 above)
Comment (anandaravuri) [reply tools/meta-bake/meta-bake.py:348]: ../fcs_prepare --hps_cert ../uboot-socfpga/u-boot-dtb.img -v 
The resulting certificate is in unsigned_cert.ccert and we need to sign it
ccert is correct

File: tools/meta-bake/n6000/quick-start-n6000.md [hunk 6]
```diff
@@ -58,6 +58,30 @@ Image  linux.dtb  rootfs.cpio
...
+* Creating User Keys using uartus_sign
+``` 
+quartus_sign --family=agilex --operation=make_private_pem --curve=secp384r1 --no_passphrase  userkey_root_private.pem
+quartus_sign --family=agilex --operation=make_public_pem  userkey_root_private.pem  userkey_root_public.pem
+quartus_sign --family=agilex --operation=make_root userkey_root_public.pem  userkey_root_public.qky 
+```
+
+
+* Copy user keys to folder meta-opae-fpga/tools/meta-bake/n6000/vab/
+* Run VAB enabled build command signs Images with user keys and generates VAB enabled Images
```
Comment (fpgamatt) [inline tools/meta-bake/n6000/quick-start-n6000.md]: Run VAB enabled build command to sign Images with user keys and generate VAB enabled Images.

File: tools/meta-bake/n6000/quick-start-n6000.md [hunk 7]
```diff
@@ -58,6 +58,30 @@ Image  linux.dtb  rootfs.cpio
...
+quartus_sign --family=agilex --operation=make_private_pem --curve=secp384r1 --no_passphrase  userkey_root_private.pem
+quartus_sign --family=agilex --operation=make_public_pem  userkey_root_private.pem  userkey_root_public.pem
+quartus_sign --family=agilex --operation=make_root userkey_root_public.pem  userkey_root_public.qky 
+```
+
+
+* Copy user keys to folder meta-opae-fpga/tools/meta-bake/n6000/vab/
+* Run VAB enabled build command signs Images with user keys and generates VAB enabled Images 
+
+ Example VABBuild Commad
```
Comment (fpgamatt) [inline tools/meta-bake/n6000/quick-start-n6000.md]: Example VAB Enabled Build Command

**Review by fpgamatt — APPROVED** [review]

**Pull request merged**